syntax="proto2";

package yail.pubsub.detail.messages;
option optimize_for = LITE_RUNTIME;

message pubsub_header
{
	enum Version
	{
		VERSION_1 = 1;
	}
	enum Type
	{
		DATA = 1; // Publisher -> Subscriber(s) (multicast)
	}
	required Version version = 1 [default = VERSION_1];
	required Type type = 2;
	required uint32 id = 3;
}

message pubsub_data
{
	required string domain = 1;
	required string topic_name = 2;
	required string topic_type_name = 3;
	required bytes topic_data = 4;
}

message pubsub
{
	required pubsub_header header = 1;
	optional pubsub_data data = 2;
}

message subscription
{
	required string domain = 1;
	required string topic_name = 2;
	required string topic_type_name = 3;
}
